include/validation/validation_service.js

```diff
@@ -151,6 +154,20 @@ ValidationService.validateVersionNum = function(value, required) {
 };
 
 /**
+ * Validates a version expression
+ *
+ * @method isVersionExpression
+ * @param {String} expression
+ * @param {Boolean} required
+ */
+ValidationService.isVersionExpression = function(expression, required) {
+  if (!expression && !required) {
+    return true;
+  }
+  return semver.valid(expression) !== null;
```

Comment:
How did you test this?   This failed right away when I began testing.  My initial understanding was that the "valid" function validated an expression.  That assumption was incorrect.  The "valid" function only validates a version.  In order to validate the expression we need to update the line to: ```return semver.validRange(expression) !== null;```